### ISSUE TITLE
Improve difficulty adjustment algorithm

### ISSUE THREAD

**Issue by macterra** [cryptotechguru/Cryptonomicon#44]
Tesseract currently uses the [Dark Gravity Wave](https://medium.com/@tronblack/ravencoin-dark-gravity-wave-1da0a71657f7) difficulty adjustment algorithm, adapted from RavenCoin. It seems to do a decent job of adjusting the difficulty to changes in hashpower but it does display a fair amount of volatility when hashpower remains constant.

![image](https://user-images.githubusercontent.com/434644/58487384-7b190d80-8135-11e9-8f55-ccf11fce8d15.png)

It is possible that replacing the algorithm with a [discrete PID controller](https://www.mathworks.com/help/control/ug/discrete-time-proportional-integral-derivative-pid-controller.html) would improve the behavioral characteristics.

**Comment by Flaxscrip** [comment]
Applying a PID controller algorithm to mining difficulty might be a first in the crypto space. Are there small steps we could do to test the concept?

**Comment by macterra** [comment]
Yes, the first step would be to develop a block mining simulation that could be used to easily test different PID parameters. The effort is likely on the order of a couple days to program a barebones prototype in python.
